UniversalAlgebra/Structures/Homs/Records.lagda

```diff
@@ -9,86 +9,71 @@ author: [agda-algebras development team][]
-open import Function.Base          using    ( _∘_   ;  id       )
-open import Relation.Binary        using    ( IsEquivalence     )
+open import Agda.Builtin.Equality  using ( _≡_ ; refl )
+open import Agda.Primitive         using ( _⊔_ ; lsuc ) renaming ( Set to Type )
+open import Data.Product           using ( _,_ ; Σ ; _×_ ; Σ-syntax ) renaming ( proj₁ to fst ; proj₂ to snd )
+open import Level                  using ( Level ;  Lift ; lift ; lower ) renaming ( zero  to ℓ₀ )
+open import Function.Base          using ( _∘_ ; id )
+open import Relation.Binary        using ( IsEquivalence )
 open import Relation.Binary.PropositionalEquality
-                                   using    ( cong  ; module ≡-Reasoning
-                                            ; sym   ; trans )
-
+                                   using ( cong ; module ≡-Reasoning ; sym ; trans )
 
 
-
-open import Overture.Preliminaries    using ( ℓ₁ ; ∣_∣ ; ∥_∥ ; _⁻¹ ; _∙_ ; 𝑖𝑑 ; Π ; Π-syntax)
-open import Overture.Inverses         using ( IsInjective ; IsSurjective ; Image_∋_)
-open import Relations.Discrete        using ( ker ; kerlift )
-open import Relations.Quotients       using ( Equivalence ; Quotient
-                                            ; 0[_]Equivalence ; ker-IsEquivalence
-                                            ; kerlift-IsEquivalence ; ⟪_⟫ ; ⌞_⌟
-                                            ; ⟪_∼_⟫-elim ; _/_ )
-open import Relations.Extensionality  using ( swelldef )
-open import Structures.Records        using ( signature ; structure ; Lift-struc ; compatible )
-open import Structures.Examples       using ( Sig∅ )
+open import Overture.Preliminaries   using ( ℓ₁ ; ∣_∣ ; ∥_∥ ; _⁻¹ ; _∙_ ; 𝑖𝑑 ; Π ; Π-syntax)
+open import Overture.Inverses        using ( IsInjective ; IsSurjective ; Image_∋_)
+open import Relations.Discrete       using ( ker ; kerlift )
+open import Relations.Quotients      using ( Equivalence ; Quotient ; 0[_]Equivalence
+                                           ; ker-IsEquivalence ; kerlift-IsEquivalence
+                                           ; ⟪_⟫ ; ⌞_⌟ ; ⟪_∼_⟫-elim ; _/_ )
+open import Relations.Extensionality using ( swelldef )
+open import Structures.Records       using ( signature ; structure ; Lift-Struc
+                                           ; Lift-Strucˡ ; compatible ; siglev₁)
+open import Structures.Examples      using ( Sig∅ )
 
 
 open structure
 open signature
 
-private variable 𝐹 𝑅 : signature
+private variable
+ 𝓞₀ 𝓥₀ 𝓞₁ 𝓥₁ : Level
+ 𝐹 : signature 𝓞₀ 𝓥₀
+ 𝑅 : signature 𝓞₁ 𝓥₁
+ α ρᵃ β ρᵇ γ ρᶜ : Level
+
 
-module _ {α ρᵃ : Level}
-         (𝑨 : structure 𝐹 {α}𝑅 {ρᵃ})
-         {β ρᵇ : Level}
-         (𝑩 : structure 𝐹 {β}𝑅 {ρᵇ}) where
+module _ (𝑨 : structure 𝐹 𝑅 {α}{ρᵃ})
+         (𝑩 : structure 𝐹 𝑅 {β}{ρᵇ}) where
 
  private
   A = carrier 𝑨
   B = carrier 𝑩
 
- preserves : (symbol 𝑅) → (A → B) → Type (α ⊔ ρᵃ ⊔ ρᵇ)
+ preserves : (symbol 𝑅) → (A → B) → Type _ -- (α ⊔ ρᵃ ⊔ ρᵇ)
```

Comment:
I too have gone back and forth on this. I've settled on: if the Level is completely obvious (to the reader, not Agda), then `_` is fine. Otherwise, I put it in explicitly. It also prevents me from doing bad refactors later, as Agda will scream.

Reply:
Sounds like a good rule of thumb.  Thanks for the tip.

Reply:
fixed